library/src/main/java/ai/promoted/metrics/usecases/TrackViewUseCase.kt

```diff
@@ -55,65 +50,49 @@ internal class TrackViewUseCase(
             return@monitored
         }
 
-        if (key != currentKey) {
-            viewId.advance()
-            currentKey = key
-        }
+        // The view was already logged, so skip it
+        if (key == currentKey) return@monitored
+
+        autoViewId.advance()
+        currentKey = key
 
         logger.enqueueMessage(
-            createViewMessage(
+            createAutoViewMessage(
                 clock = clock,
-                viewId = viewId.currentValueOrNull,
+                deviceInfoProvider = deviceInfoProvider,
+                autoViewId = autoViewId.currentValueOrNull,
                 sessionId = sessionUseCase.sessionId.currentValueOrNull,
-                name = key,
-                deviceMessage = deviceMessage
+                name = key
             )
         )
     }
 
     /**
-     * If needed (if this [key] is different than the last visible key), generates a new view ID.
-     * Then logs a view message via [MetricsLogger].
+     * Directly logs a view event using the given ID
      */
-    internal fun onImplicitViewVisible(key: String) = xray.monitored {
-        if (viewId.isOverridden) {
-            systemLogger.e(
-                IllegalStateException(
-                    "Attempted to log a new view after " +
-                            "overriding view ID"
-                )
-            )
-            return@monitored
-        }
-
-        // The view was already logged, so skip it
-        if (key == currentKey) return@monitored
-
-        viewId.advance()
-        currentKey = key
-
+    fun logView(viewId: String) = xray.monitored {
         logger.enqueueMessage(
             createViewMessage(
                 clock = clock,
-                viewId = viewId.currentValueOrNull,
+                deviceInfoProvider = deviceInfoProvider,
+                viewId = viewId,
                 sessionId = sessionUseCase.sessionId.currentValueOrNull,
-                name = key,
-                deviceMessage = deviceMessage
+                name = viewId
             )
         )
     }
 
-    fun logView(viewId: String) = xray.monitored {
-        // We don't actually need this overridden per se, but what it will do is prevent any
-        // further usage of onViewVisible, so as to avoid conflicts of view ID strategy
-        this.viewId.override(viewId)
+    /**
+     * Directly logs an auto-view event using the given ID
+     */
+    fun logAutoView(autoViewId: String) = xray.monitored {
         logger.enqueueMessage(
-            createViewMessage(
+            createAutoViewMessage(
                 clock = clock,
-                viewId = viewId,
+                deviceInfoProvider = deviceInfoProvider,
+                autoViewId = autoViewId,
                 sessionId = sessionUseCase.sessionId.currentValueOrNull,
-                name = viewId,
-                deviceMessage = deviceMessage
+                name = autoViewId
```

Comment:
The name for the view could still be a human-readable name. For example, the class name of the Activity.